src/App.tsx

```diff
@@ -1,9 +1,20 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [visibleMovies, setMovies] = useState(moviesFromServer);
+
+  const filterByQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const query = event.target.value.trim(); // trim due to test 15, "should work if user types extra spaces before or after the query"
+
+    setMovies(moviesFromServer.filter((movie) => (
+      movie.title.toLowerCase().includes(query.toLowerCase())
+      || movie.description.toLowerCase().includes(query.toLowerCase())
+    )));
```

Comment:
lets destructure movie and you can put it to separate function to no rewrite logic